### ISSUE TITLE
Add ability to install custom workload manifests as nuget packages.

### ISSUE THREAD

**Issue by AraHaan** [dotnet/sdk#21708]
### Is your feature request related to a problem? Please describe.

I would like the ability to install custom workload manifests as nuget packages using ``dotnet workload manifest install <manifest nuget package name>`` where then it will then be downloaded and unpacked into the ``dotnet/sdk-manifests/<.NET SDK version>/`` folder to the currently used .NET SDK version is used for it where it can then be installed using ``dotnet workload install <workload name>`` as if it was a valid workload manifest that would come shipped with the .NET SDK (so it can allow 3rd party workloads to be made for 3rd parties with special runtimes or special toolings).

### Describe the solution you'd like

I think allowing such would be great, it would allow my current class libs (that can in a way act as a framework and is intended to be a framework (that I want to be used with a ``FrameworkReference`` in project files) to then be made possible thanks to then allowing the installation of 3rd party workloads. Currently the only option I got is to hack it in by making a .NET global tool that basically would need to try to locate the folder where workload manifests should go in and create a new folder and copy the actual workload manifest data to the 3rd party workloads to it that way, and then run ``dotnet workload install`` on it to install it.

### Additional context

Assume company A makes their own runtime with their own apis that depends on the normal .NET 6 (or newer) runtime (a completely custom runtime similar to how WindowsDesktop has it's own runtime on top of the normal one), now they want to make it into a workload so that way their runtime does not get copied when doing builds that are *not* self contained, the way they could do that is by making their own workload manifest that then users wanting to use it to develop programs would need to install first before trying to install their workload. Now company B wants to use the runtime made by company A with their framework dependent .NET 6 (or newer) application to power most of it, they would then be able to easily install the workload manifest and their workload and it would make people in company B's lives simpler.

**Comment by mattleibow** [comment]
This is very important for platforms like Tizen that actually have all the bits in place for a working system, except that there is no way to teach .NET about the new workload ID.

Right now they have a script which does not fit so nicely with the system: https://raw.githubusercontent.com/Samsung/Tizen.NET/main/workload/scripts/workload-install.ps1

**Comment by mattleibow** [comment]
@rookiejava maybe this is what you need, or is there additional steps that need to be done?

**Comment by AraHaan** [comment]
> This is very important for platforms like Tizen that actually have all the bits in place for a working system, except that there is no way to teach .NET about the new workload ID.
> 
> Right now they have a script which does not fit so nicely with the system: https://raw.githubusercontent.com/Samsung/Tizen.NET/main/workload/scripts/workload-install.ps1

Unless you have it scan for if the ID they enter is in NuGet.org and have the package name of ``<ID>.WorkloadManifest`` and if the package type is ``DotNetPlatformWorkloadManifest`` (&& signed digitally?)?

That would also eliminate the need to store the manifests inside of the SDK installer as well which would also be great.

**Comment by JoonghyunCho** [comment]
yes, now we are providing a [nuget](https://www.nuget.org/packages/Samsung.NET.Sdk.Tizen.Manifest-6.0.300/7.0.100-preview.13.30) and a script to install a workload, and this is absolutely a better way that fits with the system if we could solve teaching .NET  about tizen workload ID.  @rookiejava what do you think?

**Comment by rookiejava** [comment]
> yes, now we are providing a [nuget](https://www.nuget.org/packages/Samsung.NET.Sdk.Tizen.Manifest-6.0.300/7.0.100-preview.13.30) and a script to install a workload, and this is absolutely a better way that fits with the system if we could solve teaching .NET about tizen workload ID. @rookiejava what do you think?

Totally agreed.
